examples/summarization/bart/run_bart_sum.py

```diff
@@ -0,0 +1,284 @@
+import argparse
+import glob
+import logging
+import os
+import random
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+from torch.utils.data import DataLoader
+
+from transformers import AdamW, BartConfig, BartTokenizer, get_linear_schedule_with_warmup
+from transformers.modeling_bart import BartForConditionalGeneration
+from utils import CnnDailyMailDataset, add_generic_args
+
+
+logger = logging.getLogger(__name__)
+
+
+class BartSystem(pl.LightningModule):
+    def __init__(self, hparams):
+        super(BartSystem, self).__init__()
+        self.hparams = hparams
+
+        config = BartConfig.from_pretrained(
+            self.hparams.config_name if self.hparams.config_name else self.hparams.model_name_or_path,
+            cache_dir=self.hparams.cache_dir if self.hparams.cache_dir else None,
+        )
+        tokenizer = BartTokenizer.from_pretrained(
+            self.hparams.tokenizer_name if self.hparams.tokenizer_name else self.hparams.model_name_or_path,
+            cache_dir=self.hparams.cache_dir if self.hparams.cache_dir else None,
+        )
+        model = BartForConditionalGeneration.from_pretrained(
+            self.hparams.model_name_or_path,
+            from_tf=bool(".ckpt" in self.hparams.model_name_or_path),
+            config=config,
+            cache_dir=self.hparams.cache_dir if self.hparams.cache_dir else None,
+        )
+        self.config, self.tokenizer, self.model = config, tokenizer, model
+
+    def forward(
+        self, input_ids, attention_mask=None, decoder_input_ids=None, decoder_attention_mask=None, lm_labels=None
+    ):
+        return self.model(
+            input_ids,
+            attention_mask=attention_mask,
+            decoder_input_ids=decoder_input_ids,
+            decoder_attention_mask=decoder_attention_mask,
+            lm_labels=lm_labels,
+        )
+
+    def _step(self, batch):
+        y = batch["target_ids"]
+        lm_labels = y.clone()
+        lm_labels[y == self.tokenizer.pad_token_id] = -100
+
+        outputs = self(
+            batch["source_ids"],
+            attention_mask=batch["source_mask"],
+            decoder_input_ids=batch["target_ids"],
+            lm_labels=lm_labels,
+        )
```

Comment:
This asks the model to predict the input IDs, which leads to a loss of 0

I think you need:
```suggestion
        y_ids = y[:,:-1].contihuous()
        lm_labels = y[:,1:].clone()
        lm_labels[y[:,1:] == self.tokenizer.pad_token_id] = -100

        outputs = self(
            batch["source_ids"],
            attention_mask=batch["source_mask"],
            decoder_input_ids=y_ids,
            lm_labels=lm_labels,
        )
```

Reply:
Good catch, thanks!